fili-core/src/main/java/com/yahoo/bard/webservice/web/SlicesApiRequest.java

```diff
@@ -174,18 +190,47 @@ protected SlicesApiRequest() {
                         }
                 );
 
+        Set<SortedMap<DateTime, Map<String, SegmentInfo>>> sliceMetadata = dataSourceMetadataService.getTableSegments(
+                Collections.singleton(table)
+        );
+
         Map<String, Object> generated = new LinkedHashMap<>();
         generated.put("name", sliceName);
         generated.put("timeGrain", table.getTimeGrain().getName());
         generated.put("timeZone", table.getTimeGrain().getTimeZoneName());
         generated.put("dimensions", dimensionsResult);
         generated.put("metrics", metricsResult);
+        generated.put("segmentInfo", generateSegmentMetadataView(sliceMetadata));
 
         LOG.trace("Generated slice: {}", generated);
 
         return generated;
     }
 
+    /**
+     * Create a simplifying view of segment info for a particular slice.
+     *
+     * @param sliceMetadata  The raw set of maps of datetime to maps of segment names to segment details
+     *
+     * @return A simplified Map of dateTime to set of segment identifiers
+     */
+    private static Map<DateTime, Set<String>> generateSegmentMetadataView(
+            Set<SortedMap<DateTime, Map<String, SegmentInfo>>> sliceMetadata
+    ) {
+        Map<DateTime, Set<String>> segmentView;
+        if (!sliceMetadata.isEmpty()) {
+            SortedMap<DateTime, Map<String, SegmentInfo>> segmentInfo = sliceMetadata.iterator().next();
```

Comment:
If this is handled, I _think_ the conditional around `!sliceMetadata.isEmpty()` can go away too, since a stream over the top-level set (if empty) would produce an empty Map anyways. I'm always a fan of simplifying conditionals and special-cases